src/MongoDB/Manager.c

```diff
@@ -798,7 +798,13 @@ static void php_phongo_manager_free_object(zend_object* object) /* {{{ */
 	zend_object_std_dtor(&intern->std);
 
 	if (intern->client) {
-		MONGOC_DEBUG("Not destroying persistent client for Manager");
+		if (intern->use_persistent_client) {
+			MONGOC_DEBUG("Not destroying persistent client for Manager");
+		} else {
+			MONGOC_DEBUG("Destroying private client for Manager");
+			mongoc_client_destroy(intern->client);
```

Comment:
We've never had to deal with this before, but I think actually need test(s) to ensure that this isn't ever called before other resources are freed. Any struct that hold a `mongoc_client_t*` will likely need to be changed to hold a reference to the Manager object to ensure that this object isn't garbage-collected before those dependent objects are freed.

This should be straightforward for most things (e.g. WriteResult, Cursor, Session, Server), but I think the command monitoring events will be trickier. Currently, they hold a client reference so that they can construct Server objects on-demand. You'd need to refactor that so they hold an actual Server object, which in turn has a reference to a Manager. One problem is that you won't know what Manager to refer to when doing this for persistent clients. We presumably already have this issue today, but the Server objects we create are totally independent from a Manager so it never comes up. That said, this referencing-holding is only need for non-persistent Managers so maybe it doesn't matter how you handle this (or if you do at all, assuming you want to support two code paths).